common/src/androidTest/java/com/microsoft/identity/common/internal/migration/AdalMigrationAdapterDeserializationTest.java

```diff
@@ -0,0 +1,139 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.migration;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import com.google.gson.Gson;
+import com.microsoft.identity.common.internal.cache.ADALTokenCacheItem;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class AdalMigrationAdapterDeserializationTest {
```

Comment:
Does it need to be an instrumented test? Wouldn't a JVM test be more efficient?

Reply:
Need a context

Reply:
You can get a context in JVM based tests. `ApplicationProvider.getApplicationContext()` - it's an AndroidX API

All of our E2E Robolectric tests need a `context` as well and we get it like that and run the tests in JVM

Reply:
Instrumented test is fine too....it's just that we've been trying to move to Robolectric test as a team, with possibly moving our existing Instrumented test to Robolectric in the future....adding more instrumented tests would just make that more difficult to happen

Reply:
e143f6dbbdd0787210be518e3bfdda4180635ae6